src/System.Private.CoreLib/src/System/Collections/Generic/EqualityComparer.cs

```diff
@@ -315,7 +315,7 @@ internal override int LastIndexOf(byte[] array, byte value, int startIndex, int
     [Serializable]
     [System.Runtime.CompilerServices.TypeForwardedFrom("mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089")]
     // Needs to be public to support binary serialization compatibility
-    public sealed class EnumEqualityComparer<T> : EqualityComparer<T>, ISerializable where T : struct
+    public sealed class EnumEqualityComparer<T> : EqualityComparer<T>, ISerializable where T : struct, Enum
```

Comment:
It's "public" but not actually considered public / in the contracts, which is why it's ok to augment the constraint?